File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/GraphDefinition.ts
import { UsageContext } from "./UsageContext";
import { ContactDetail } from "./ContactDetail";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { BackboneElement } from "./BackboneElement";

export type GraphDefinitionLinkTargetCompartment = BackboneElement & {
    use: string;
    code: string;
    rule: string;
    expression?: string;
    description?: string;
};

export type GraphDefinitionLinkTarget = BackboneElement & {
    link?: Reference[];
    type: string;
    params?: string;
    profile?: string;
    compartment?: GraphDefinitionLinkTargetCompartment[];
};

export type GraphDefinitionLink = BackboneElement & {
    max?: string;
    min?: number;
    path?: string;
    target?: GraphDefinitionLinkTarget[];
    sliceName?: string;
    description?: string;
};

export type GraphDefinition = DomainResource & {
    description?: string;
    date?: string;
    publisher?: string;
    jurisdiction?: CodeableConcept[];
    purpose?: string;
    name: string;
    start: string;
    useContext?: UsageContext[];
    experimental?: boolean;
    status: string;
    link?: GraphDefinitionLink[];
    url?: string;
    version?: string;
    contact?: ContactDetail[];
    profile?: string;
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/GraphDefinition.py
from typing import Optional, List
from pydantic import *
from ..base import *

class GraphDefinition_Link_Target_Compartment(BackboneElement):
    use: str
    code: str
    rule: str
    expression: Optional[str] = None
    description: Optional[str] = None

class GraphDefinition_Link_Target(BackboneElement):
    link: Optional[List[Reference]] = None
    type: str
    params: Optional[str] = None
    profile: Optional[str] = None
    compartment: Optional[List[GraphDefinition_Link_Target_Compartment]] = None

class GraphDefinition_Link(BackboneElement):
    max: Optional[str] = None
    min: Optional[int] = None
    path: Optional[str] = None
    target: Optional[List[GraphDefinition_Link_Target]] = None
    slice_name: Optional[str] = None
    description: Optional[str] = None

class GraphDefinition(DomainResource):
    description: Optional[str] = None
    date: Optional[str] = None
    publisher: Optional[str] = None
    jurisdiction: Optional[List[CodeableConcept]] = None
    purpose: Optional[str] = None
    name: str
    start: str
    use_context: Optional[List[UsageContext]] = None
    experimental: Optional[bool] = None
    status: str
    link: Optional[List[GraphDefinition_Link]] = None
    url: Optional[str] = None
    version: Optional[str] = None
    contact: Optional[List[ContactDetail]] = None
    profile: Optional[str] = None